### PR TITLE
Patroni 2.1.4

### PR DESCRIPTION
**Pull request by CyberDem0n**
- ensure all packages are upgraded to the latest version (apt-get -y upgrade)
- allow starting etcd on non-x86 platforms
- refactor `get_dcs_config()` function

Close https://github.com/zalando/spilo/issues/736, https://github.com/zalando/spilo/issues/726

### CODE REVIEW

File: postgres-appliance/scripts/configure_spilo.py [hunk 1]
```diff
@@ -689,39 +697,38 @@ def pystache_render(*args, **kwargs):
 
 
 def get_dcs_config(config, placeholders):
+    # (KUBERNETES|ZOOKEEPER|EXHIBITOR|CONSUL|ETCD3|ETCD)_(HOSTS|HOST|PORT|...)
+    dcs_configs = defaultdict(dict)
+    for name, value in placeholders.items():
+        if '_' not in name:
+            continue
+        dcs, param = name.lower().split('_', 1)
+        if dcs in PATRONI_DCS:
```
Comment (sdudoladov) [inline postgres-appliance/scripts/configure_spilo.py:706]: should not that raise a warning if the condition does not hold ?

File: postgres-appliance/scripts/configure_spilo.py [hunk 2]
```diff
@@ -647,9 +646,18 @@ def get_placeholders(provider):
...
+    cgroup_v2_memory_limit_path = '/sys/fs/cgroup/memory.max'
+
+    if os.path.exists(cgroup_memory_limit_path):
+        with open(cgroup_memory_limit_path) as f:
             os_memory_mb = int(f.read()) / 1048576
+    elif os.path.exists(cgroup_v2_memory_limit_path):
+        with open(cgroup_v2_memory_limit_path) as f:
+            try:
+                os_memory_mb = int(f.read()) / 1048576
+            except Exception:  # there could be "max"
```
Comment (sdudoladov) [inline postgres-appliance/scripts/configure_spilo.py:659]: what does the phrase 
>  there could be "max"

mean here ?

To me the code reads like `If there is any exception, pretend there is no limit on os_memory_mb by setting it to the max possible value`

File: postgres-appliance/scripts/configure_spilo.py (hunk 1 above)
Comment (CyberDem0n) [reply postgres-appliance/scripts/configure_spilo.py:706]: No, there could be many different environment variables, this is absolutely normal.

File: postgres-appliance/scripts/configure_spilo.py (hunk 2 above)
Comment (sdudoladov) [reply postgres-appliance/scripts/configure_spilo.py:659]: oh now I understand.
can you change the comment to 

>  string literal "max" is a possible value

?

**Comment by FxKu** [comment]
:+1:

**Comment by CyberDem0n** [comment]
:+1:

**Pull request merged**